### PR TITLE
Compare tags using common tag type

### PR DESCRIPTION
**Pull request by RedbackThomson**
Implements https://github.com/aws-controllers-k8s/community/issues/1658

Description of changes:
Instead of comparing tag fields using the standard `reflect.DeepEquals`, this PR updates the code-generator to convert the tags to the common `acktags.Tag` type and then compare those as maps. This ensures that changes in ordering do not result in false positive deltas.

Controllers that rely on custom tag delta functions also typically use a function to determine which tags are being added and which are being removed. Using the same trick as this PR, a method can be added to the common runtime to deduce those - resulting in far less duplicated and verbose custom hook code. Those changes are not included in this PR, though.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/generate/code/compare_test.go [hunk 1]
```diff
@@ -333,12 +340,8 @@ func TestCompareResource_Lambda_Function(t *testing.T) {
 			delta.Add("Spec.Runtime", a.ko.Spec.Runtime, b.ko.Spec.Runtime)
 		}
 	}
-	if ackcompare.HasNilDifference(a.ko.Spec.Tags, b.ko.Spec.Tags) {
+	if !ackcompare.MapStringStringEqual(ToACKTags(a.ko.Spec.Tags), ToACKTags(b.ko.Spec.Tags)) {
```
[the remark this answers is not in the document]
Comment (RedbackThomson) [reply pkg/generate/code/compare_test.go:343]: It's part of the generated code within each resource https://github.com/aws-controllers-k8s/ec2-controller/blob/main/pkg/resource/vpc/tags.go

[the remark this answers is not in the document]
Comment (jaypipes) [reply pkg/generate/code/compare_test.go:343]: Oh, duh, yeah I forgot about that. I thought for a second it was in `runtime`'s `pkg/tags`

**Review by jaypipes — APPROVED** [review]
:+1: this is great, @RedbackThomson thank you!

**Review by A-Hilaly — APPROVED** [review]
To the moooooon 🌔

File: pkg/testdata/models/apis/lambda/0000-00-00/generator.yaml [hunk 2]
```diff
@@ -25,3 +25,6 @@ resources:
           in:
             - 1
             - 2
+  CodeSigningConfig:
+    tags:
+      ignore: true
```
Comment (jljaco) [inline pkg/testdata/models/apis/lambda/0000-00-00/generator.yaml:30]: For my own edification:  why is this necessary here for `Lambda.CodeSigningConfig` but not for other applicable test cases like `MemoryDB.User`?

Comment (RedbackThomson) [reply pkg/testdata/models/apis/lambda/0000-00-00/generator.yaml:30]: `CodeSigningConfig` doesn't support tags in its create call - https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#CreateCodeSigningConfigInput - and for the purposes of these tests we aren't adding the custom field to support it
